glean-core/ios/GleanTests/Metrics/TimespanMetricTests.swift

```diff
@@ -0,0 +1,178 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+@testable import Glean
+import XCTest
+
+class TimespanMetricTypeTests: XCTestCase {
+    override func setUp() {
+        Glean.shared.resetGlean(clearStores: true)
+    }
+
+    func testTimespanSavesToStorage() {
+        let metric = TimespanMetricType(
+            category: "telemetry",
+            name: "timespan_metric",
+            sendInPings: ["store1"],
+            lifetime: .application,
+            disabled: false,
+            timeUnit: .millisecond
+        )
+
+        XCTAssertFalse(metric.testHasValue())
+
+        // Record a timespan
+        metric.start()
+        metric.stop()
+
+        // Check that the count was incremented and properly recorded.
+        XCTAssert(metric.testHasValue())
+        XCTAssert(try metric.testGetValue() >= 0)
+    }
+
+    func testTimespanMustNotRecordIfDisabled() {
+        let metric = TimespanMetricType(
+            category: "telemetry",
+            name: "timespan_metric",
+            sendInPings: ["store1"],
+            lifetime: .application,
+            disabled: true,
+            timeUnit: .millisecond
+        )
+
+        // Record a timespan.
+        metric.start()
+        metric.stop()
+
+        // Let's also call cancel() to make sure it's a no-op.
+        metric.cancel()
+
+        XCTAssertFalse(metric.testHasValue(), "Timespan must not be recorded if they are disabled")
+    }
+
+    func testTimespanMutCorrectlyCancel() {
+        let metric = TimespanMetricType(
+            category: "telemetry",
+            name: "timespan_metric",
+            sendInPings: ["store1"],
+            lifetime: .application,
+            disabled: false,
+            timeUnit: .millisecond
+        )
+
+        // Record a timespan.
+        metric.start()
+        metric.cancel()
+        metric.stop()
+
+        XCTAssertFalse(metric.testHasValue(), "Timespan must not be recorded if they are disabled")
+    }
+
+    func testTimespanGetValueThrowsExceptionIfNothingIsStored() {
+        let metric = TimespanMetricType(
+            category: "telemetry",
+            name: "timespan_metric",
+            sendInPings: ["store1"],
+            lifetime: .application,
+            disabled: false,
+            timeUnit: .millisecond
+        )
+
+        do {
+            _ = try metric.testGetValue()
```

Comment:
There is also an [`XCTAssertThrowsError`](https://developer.apple.com/documentation/xctest/1500795-xctassertthrowserror) function that may be useful here

Reply:
for _some_ reason I didn't find that before oO